documentation/design-docs/dotnet-monitor.yaml

```diff
@@ -0,0 +1,457 @@
+      parameters:
+      - name: "type"
+        in: query
+        description: "Specifies the dump type, which determines the kinds of information that are collected from the process."
+        type: string
+        enum:
+        - "Normal"
+        - "WithHeap"
+        - "Triage"
+        - "Full"
+        required: false
+        default: "Normal"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /dump/{uid}:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getprocessdump"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "uid"
+        in: "path"
+        description: "Process ID of the process to collect a process dump"
+        required: true
+        type: "integer"
+      - name: "type"
+        in: query
+        description: "Specifies the dump type, which determines the kinds of information that are collected from the process."
+        type: string
+        enum:
+        - "Normal"
+        - "WithHeap"
+        - "Triage"
+        - "Full"
+        required: false
+        default: "Normal"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /gcdump/:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getdefaultgcdump"
+      produces:
+      - "application/octet-stream"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /gcdump/{uid}:
+    get:
+      tags:
+      - "dumps"
+      operationId: "getgcdump"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "uid"
+        in: "path"
+        description: "Process ID of the process to collect a process dump"
+        required: true
+        type: "string"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /cpuprofile:
+    get:
+      tags:
+      - "traces"
+      operationId: "getdefaultcputrace"
+      produces:
+      - "application/octet-stream"
+      parameters:
+      - name: "duration"
+        in: "query"
+        description: "Duration of the CPU trace"
+        required: false
+        type: "integer"
+        default: "30"
+      responses:
+        200:
+          description: "Success"
+          schema:
+            format: "byte"
+        400:
+          description: "Bad request"
+          schema:
+            $ref: "#/definitions/Error"
+  /cpuprofile/{uid}:
```

Comment:
Do we want /operation/{pid} or /{pid}/operation? I'm not sure what is more conventional.

Reply:
According to REST guidelines, it's noun/{id}. URIs should not contain the operation name.

How I've interpreted it here is in `GET /cpuprofile/{uid}`, I'm retrieving the CPUProfile resource with Id = PID.

I don't have a preference, but neither really conform to idiomatic REST since we're putting the operation in the URI.